File: Backend/questao_2.py
from collections import Counter
def q2_contar_frequencia_palavra(text):
    """Realiza uma contagem de quantas vezes uma determinada palavra ocorre
    Ao final, preferencialmente, ordenar o dicionário pelo volume de ocorrência

    Saída esperada: {'hello': 2, 'world': 1}
    """
    control = text.lower().split(' ')
    controlSet = set(control)
    sorting_list = []
    maior = 0
    menor =  0

    dictionary ={}

    for word in controlSet:
        if (num := control.count(word)) > menor:

            if num < maior:
                sorting_list.append(word)

            elif num > maior:
                maior = num
                sorting_list.insert(0, word)

    for name in sorting_list:
        dictionary[name] = control.count(name)
        

    return dictionary

print(q2_contar_frequencia_palavra("lucas Hello world hello lucas lucas lucas"))

File: Frontend/questao_2.js
function q2ContarFrequenciaPalavra(text) {
    const control = text.toLowerCase().split(' ');
    const controlSet = new Set(control);
    const sortingList = [];
    let maior = 0;
    let menor = 0;
  
    const dictionary = {};
  
    for (const word of controlSet) {
      const num = control.filter(item => item === word).length;
  
      if (num > menor) {
        if (num < maior) {
          sortingList.push(word);
        } else if (num > maior) {
          maior = num;
          sortingList.unshift(word);
        }
      }
    }
  
    for (const name of sortingList) {
      dictionary[name] = control.filter(item => item === name).length;
    }
  
    return dictionary;
  }
console.log(q2ContarFrequenciaPalavra("Hello world hello"));